### ISSUE TITLE
[Bug]: Filter Function Not Working When Language is Set to Japanese

### ISSUE THREAD

**Issue by Daiki-Nakamura-git** [jasp-stats/jasp-issues#2971]
### JASP Version

0.19.1

### JASP Module

Unrelated

### What analysis are you seeing the problem on?

The Click Filter

### What OS are you seeing the problem on?

Windows 11

### Bug Description

When the language setting is switched to Japanese in JASP, the filter function, which selects data based on specific variable values, does not work as expected. This issue does not occur when the language is set to English, so it seems to be related to the language setting.

![スクリーンショット 2024-10-15 213901](https://github.com/user-attachments/assets/80bea6f5-29ff-44ff-ae19-60b294512279)

### Expected Behaviour

The filter function should allow users to select data based on specific variable values, regardless of the language setting. In the English version, the filter function works correctly and allows filtering data as intended.

### Steps to Reproduce

1. Open JASP and switch the language setting to Japanese.
2. Load a dataset.
3. Try to use the filter function to select data based on specific variable values.
4. Notice that the filter function does not work as expected.
5. Switch the language back to English, and repeat the steps.
6. Observe that the filter function works correctly in the English setting.

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [X] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by shun2wang** [comment]
Hi @Daiki-Nakamura-git I cannot reproduce it, is it only happened once or every time? Could you please upload the data file with this issue?

**Comment by Daiki-Nakamura-git** [comment]
Hi @shun2wang  When the language is set to Japanese, issues occur with any type of data. I have attached an example of the data. Since other Japanese users around me have confirmed the same issue, I believe it is reproducible.

[data1.csv](https://github.com/user-attachments/files/17390644/data1.csv)

**Comment by shun2wang** [comment]
I still can't reproduce this problem with the data you provided, even with Japanese language. open the "about" window of JASP, and click "copy debug information" and paste in github comment. you should also provide logs, see: https://github.com/jasp-stats/jasp-desktop/blob/development/Docs/user-guide/logging-howto.md

with thanks.

**Comment by tomtomme** [comment]
Also all fine on linux flatpak:
![grafik](https://github.com/user-attachments/assets/4f22b61c-bf63-4d70-87ce-2d270547f782)

On which OS did you test @shun2wang ?
maybe a win11 specific thing

**Comment by Daiki-Nakamura-git** [comment]
@shun2wang Thanks for your quick response.
This is debug information.

```
-------- Application Info --------
JASP Version: JASP 0.19.1
Build Branch: lockfiletesting
Build Date: Sep  9 2024 19:00:04 (Netherlands)
Last Commit: 8ac10c9c7aec029fdd695262d7bb4c3742a8ec89

-------- Basic Info --------
Operating System: Windows 11 Version 23H2
Product Version: 11
Kernel Type: winnt
Kernel Version: 10.0.22631
Architecture: x86_64
Install Path: C:/Program Files (x86)/JASP
Platfotm Name: windows
System Local: ja_JP

-------- Extra Info --------
Current code page
���݂̃R�[�h �y�[�W: 932
Active code page: 65001

```

And  I have attached the log file.
[JASP 2024-10-16 16_20_24 Desktop.log](https://github.com/user-attachments/files/17391068/JASP.2024-10-16.16_20_24.Desktop.log)
[JASP 2024-10-16 16_20_24 Engine 0.log](https://github.com/user-attachments/files/17391069/JASP.2024-10-16.16_20_24.Engine.0.log)

**Comment by shun2wang** [comment]
Yes I am on Windows now. seems it was fixed in new development, 
plese try: 
https://static.jasp-stats.org/Nightlies/WindowsMSIX/JASP-nightly-development-c676de286f33587cddd5f2fce2f22117a77d9631.msix

and see if it has been fixed. note this is a version under developing.

**Comment by shun2wang** [comment]
A problem with r code send to Engine, I can get same error in R.
```r
local({generatedFilter <- (JaspColumn_5_Encoded == \"m\") & \n((JaspColumn_2_Encoded == '5')\n)\n\nこのデフォルトの結果は'generatedFilter'に保存され、カスタムフィルターと置き換えたり、 組み合わせたりすることができます。\n組み合わせるには'&'を使って節を追加します: 'generatedFilter & customFilter & perhapsAnotherFilter'.\n右下の(i)アイコンをクリックすると, さらなるヘルプが表示されます.\n\ngeneratedFilter\n})
```

@JorisGoosen @vandenman any idea?

**Comment by JorisGoosen** [comment]
I think probably in the translation the `#` character is missing in the explanation of how to use the R filter code together with generatedFilter.

Ive already fixed this by forcing the translatable part to always be prefixed with '#' in https://github.com/jasp-stats/jasp-desktop/blob/937d636cc2d6eaf9a77a875ab66de43abc7262f1/Desktop/data/filtermodel.cpp#L34

**Comment by shun2wang** [comment]
So it will be included in the next release. feel free to open it if not solved later.

**Comment by Daiki-Nakamura-git** [comment]
@shun2wang  I tried to install the development version you provided, but I encountered an error stating that the signature could not be found (0x800B010A), and I was unable to complete the installation. Could you please add the signature?

**Comment by shun2wang** [comment]
Sorry that development version was not signed, please try .zip version, just unzip it and run `JASP.exe`. https://static.jasp-stats.org/Nightlies/WindowsZip/JASP-nightly-development-c676de286f33587cddd5f2fce2f22117a77d9631.zip

**Comment by JorisGoosen** [comment]
Yeah Im checking though, part of it changed the translated strings. So they will need to be retranslated, but maybe I can move some of the old translations myself.

**Comment by JorisGoosen** [comment]
Ok, ive fixed at least the japanese one and my local build now works:
<img width="1260" alt="afbeelding" src="https://github.com/user-attachments/assets/6eae6587-fa84-42c1-9e7c-fb97a4be287c">

Ill update the other translations by hand and upload it, it will take at least a night to get a new nightly probably.
(Unless you want intel mac, cause that buildsystem is currently not working well)

**Comment by Daiki-Nakamura-git** [comment]
@shun2wang After extracting the ZIP file and running the development version of JASP you provided, the filter worked without any issues even with the Japanese settings. I appreciate your efforts in making these improvements.

**Comment by hiroue80x** [comment]
The same problem seems to persist in mac version 19.1, is there any solution?

<img width="1718" alt="erro" src="https://github.com/user-attachments/assets/1ed59015-c3e0-4d8a-905d-ed88441cc4b1">

**Comment by tomtomme** [comment]
This is only fixed in Version 0.19.2 which will be out in a month or so

**Comment by shun2wang** [comment]
As a temporary solution, you can switch to the English interface and run that analysis. @hiroue80x

**Comment by hiroue80x** [comment]
Thanks for the quick response. I'll temporarily switch to the English interface. And I'm waiting for 0.19.2 to come out.
